presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java

```diff
@@ -131,6 +131,24 @@ public void run()
             injector.getInstance(Announcer.class).start();
 
             log.info("======== SERVER STARTED ========");
+
+            ServerConfig serverConfig = injector.getInstance(ServerConfig.class);
+
+            if (serverConfig.isCoordinator() && serverConfig.isEnabledUIonSecondaryPort()) {
+                Bootstrap uIApp = new Bootstrap(new NodeModule(),
+                                                new DiscoveryModule(),
+                                                new JsonModule(),
+                                                new JaxrsModule(true),
+                                                new JmxModule(),
+                                                new JmxHttpModule(),
+                                                new JsonEventModule(),
+                                                new HttpEventModule(),
+                                                new ServerSecurityModule(),
+                                                new CoordinatorUIHttpServerModule(injector));
+                Injector uIInjector = uIApp.doNotInitializeLogging().initialize();
+
+                log.info("======== SECONDARY UI STARTED ========");
```

Comment:
So basically, we cannot change the current "http-server.http.port" to a new property name like "http-server.protocal.http.port" since it is defined in airlift's HttpServerModule. Based on this restriction I would prefer the secondary UI concept instead of the concept of protocol port plus UI port.